text/0001-poet2-consensus.md

```diff
@@ -0,0 +1,109 @@
+PoET 2.0, however, removes the reliance on internal timers and monotonic counters within SGX. This necessitates a different approach for enforcing wait times. Rather than set a timer inside SGX, PoET 2.0 relies on SGX to provide it with a duration to wait and sets the timer OUTSIDE SGX. A claim block is forwarded only after the timer has expired. The block is now accompanied by a WaitCertificate from SGX containing the duration that the block was expected to wait for.
+While this opens up the possibility of misuse by a malicious actor, the community of validators is engaged to enforce the wait on the block. Blocks arriving at peers earlier than suggested by their WaitCertificate are held back until the time is right to forward them. The larger the size of the network, the more resistant the protocol is to attack by a malicious actor or a cabal.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Here we describe the PoET 2.0 algorithm in greater detail.
+
+## Initialization
+When the PoET 2.0 module is initialized, it performs the following actions:
+1. Generate an ECDSA key pair. This pair is held in SGX memory and NEVER committed to disk for reuse subsequently.
+2. Create a table in SGX memory to map the Block Number to WaitCertificates. Also store the block number of the chain head in SGX.
+3. Generate a Quote containing the ECDSA public key and perform Quote verification with the IAS.
+4. Check if there was a previous ECDSA key registration. If there was, verify if ‘K’ blocks have been added to the chain since the key’s registration. If K blocks have not passed, wait until K blocks have been added.
+5. Register the ECDSA Public key and the IAS response with the ledger. 
+6. Wait for ‘C’ blocks to be added to the chain before participating in consensus.
+
+## Wall Clock and Chain Clock
+The PoET 2.0 algorithm maintains two clocks, the _Wall Clock_ and the _Chain Clock_.
+The Wall Clock (WC) measures the time elapsed since the arrival of the ‘Sync Block’, i.e. the first block on the chain.
+The Chain Clock (CC) measures the cumulative wait duration of all the blocks in the chain (each fork will have its own CC).
+Upon receiving the Sync Block, WC and CC are initialized to 0. In practice, the WC may be calculated by recording the system time at the moment of the arrival of the Sync Block and subsequently subtracting this timestamp from the current time.
+To handle clock drifts, it may become necessary to synchronize with a standard network clock. Details are out of scope for this version of the document.
+
+## Block Creation
+Once the algorithm is in a state to publish blocks (i.e. the C test passes), it starts the process of creating claim blocks with the aim of participating in the Consensus mechanism.
```

Comment:
I found "the C test" to be confusing.  Is it the same thing as Step 6 in the previous section? ("Wait for ‘C’ blocks to be added ...").  Could you clarify here?